### ISSUE TITLE
Arch → Arch menu → Reinforcement tools

### ISSUE THREAD

**Issue by kaktusus** [FreeCAD/FreeCAD-translations#224]
```
OS: Debian GNU/Linux 12 (bookworm) (KDE/plasma)
Word size of FreeCAD: 64-bit
Version: 0.21.0.33110 (Git) AppImage
Build type: Release
Branch: master
Hash: 18e40941ceb36c52701a7c74ea26043ef3385701
Python 3.11.3, Qt 5.15.8, Coin 4.0.0, Vtk 9.2.2, OCC 7.7.1
Locale: Polish/Poland (pl_PL)
Installed mods: 
  * Curves 0.5.0
  * sheetmetal 0.2.56
  * Reinforcement
```

Tool labels are not translated.

![obraz](https://github.com/FreeCAD/FreeCAD-translations/assets/39754351/e132d606-a984-4442-be07-6d9281109f95)

1. `Straight Rebar`
2. `U-Shape Rebar`
3. `L-Shape Rebar`
4. `Stirrup`
5. `Bent Shape Rebar`
6. `Helical Rebar`
7. `Column Reinforcement`
8. `Beam Reinforcement`
9. `Slab Reinforcement`
10. `Footing Reinforcement`

all of these strings do not found in Crowdin _(FreeCAD, not freecad-addons)_
eg: https://crowdin.com/translate/freecad/all/en-pl?filter=basic&value=0#q=Beam%20Reinforcement

I also noticed an anomaly. I also noticed an anomaly. When I installed the BIM 2021.12.0 add-on, the labels of these tools show up as translated but only after switching to BIM workbench and then back to Arch workbench again.

In BIM, on the other hand, there is an untranslated sub-menu name `Reinforcement tools` - no string in Crowdin.
To be funny in Arch workbench this string is translated but I could not find it in Crowdin. :scream: 

Should I create a separate ticket for BIM wb? There is another lack of translation in addition to the one mentioned.

**Comment by chennes** [comment]
@yorikvanhavre where are these strings coming from?

**Comment by yorikvanhavre** [comment]
From the Reinforcement workbench at https://github.com/amrit3701/FreeCAD-Reinforcement
That addon is in https://crowdin.com/project/freecad-addons/sources/files bht the translations have not been updated recently. Maybe @amrit3701 can have a look?

**Comment by amrit3701** [comment]
@yorikvanhavre @kaktusus yes, reinforcement translations are lit bit old but I will update the translations in Crowdin and let you know asap. thanks.

**Comment by kaktusus** [comment]
We also have two old tickets that have not been resolved so far.

1.  https://github.com/FreeCAD/FreeCAD-translations/issues/135 

2.  https://github.com/FreeCAD/FreeCAD-translations/issues/136

**Comment by amrit3701** [comment]
@kaktusus 

I apologise for keeping you waiting so long, but it appears that all of the reinforcement translations are available.

The problem is that if I access the **Arch** workbench without first opening the 'Reinforcement' workbench, the rebar tools are not translated.

<img width="993" alt="image" src="https://github.com/FreeCAD/FreeCAD-translations/assets/10414959/eb40b466-4362-4853-8a94-2316f8d56744">



If I opened the **Reinforcement** workbench once, then opened the **Arch** workbench, the problem automatically disappeared.

<img width="1086" alt="image" src="https://github.com/FreeCAD/FreeCAD-translations/assets/10414959/e067ec8a-cb72-4e69-b1d9-389280d0b216">



@yorikvanhavre Do you have any idea to load **Reinforcement** translation along with **Arch** workbench.

**Comment by kaktusus** [comment]
yes I noticed that.
And I informed you about it at the beginning of the notification.

**Comment by yorikvanhavre** [comment]
> @yorikvanhavre Do you have any idea to load Reinforcement translation along with Arch workbench.
Are they inside a RC file? I would maybe add it here next to the Arch one, inside a `try:` statement:
https://github.com/FreeCAD/FreeCAD/blob/master/src/Mod/Arch/InitGui.py#L54

**Comment by kaktusus** [comment]
![obraz](https://github.com/FreeCAD/FreeCAD-translations/assets/39754351/9a36ed5d-7187-4777-bac2-74e6409bae9c)

It is interesting that one item is displayed correctly,
after starting the program I activate Arch wb.

**Comment by amrit3701** [comment]
> It is interesting that one item is displayed correctly,
> after starting the program I activate Arch wb.

Because `Custom Rebar` is the part of Arch wb and all other rebars are defined in Reinforcement wb.

**Comment by amrit3701** [comment]
@kaktusus Here is PR: https://github.com/FreeCAD/FreeCAD/pull/9889

We can close this ticket after the PR is merged into master.

**Comment by kaktusus** [comment]
Well, 
thank you for the information.
I will close this ticket when I see a working translation in the GUI :wink:

**Comment by kaktusus** [comment]
Thank you for your efforts.

**Comment by kaktusus** [comment]
Fixed

```
OS: Debian GNU/Linux trixie/sid (KDE/plasma)
Word size of FreeCAD: 64-bit
Version: 0.21.0.33631 (Git) AppImage
Build type: Release
Branch: master
Hash: 2167fb6f7ec6ac5c2f6365d4c3035d2db6879b1e
Python 3.10.12, Qt 5.15.8, Coin 4.0.0, Vtk 9.2.5, OCC 7.6.3
Locale: Polish/Poland (pl_PL)
Installed mods: 
  * Curves 0.5.0
  * sheetmetal 0.2.56
  * Reinforcement
  * BIM 2021.12.0
  * Rocket 3.1.0
  * kicadStepUpMod 10.19.4
  * fasteners 0.4.56
  * Alternate_OpenSCAD 1.0.0
  * Render 2023.2.6
```


![image](https://github.com/FreeCAD/FreeCAD-translations/assets/39754351/03cf3f10-5c8f-46cd-a2b7-d375bd918240)

**Comment by kaktusus** [comment]
overall, no
the error persists
I forgot that I previously activated Arch wb.

**Comment by kaktusus** [comment]
Today I performed another test.

I loaded the workbench Arch
- there was a lack of translation of reinforcement tools
I loaded the BIM workbench in turn.
Once again loaded the Arch workbench
 - this time the names of all rebar tools were displayed as translated.

**Comment by luzpaz** [comment]
@amrit3701 :point_up_2:

**Comment by amrit3701** [comment]
> overall, no the error persists I forgot that I previously activated BIM wb.
> 
> ![image](https://user-images.githubusercontent.com/39754351/255895126-3812ffa5-e3d7-45d5-9390-843696cef181.png)

@kaktusus Have you clone the latest code of Reinforcement WB from Addon manager?

https://github.com/amrit3701/FreeCAD-Reinforcement

**Comment by kaktusus** [comment]
Yes I have updated this add-on.

![obraz](https://github.com/FreeCAD/FreeCAD-translations/assets/39754351/eda77a45-8269-424f-9f9e-58a32fd7aa1e)

**Comment by amrit3701** [comment]
From screenshot it looks like you are using v0.3.0. Please clone latest `master` branch code (latest commit: `ade0085b3b1dac99afc353d246b903d814240315`).

<img width="1478" alt="image" src="https://github.com/FreeCAD/FreeCAD-translations/assets/10414959/e09e5de2-84c5-4d3e-8bc7-553354c8ae22">

BTW, I am also planning to release v0.4 with all translation fix but it will take time.

**Comment by kaktusus** [comment]
well 
I will test the new version.

Will you be doing a string import into Crowdin?

**Comment by kaktusus** [comment]
At a glance:

1.  downloaded the `master` version you indicated
2.  installed manually by extracting the archive to the `mod` directory
3.  launched the add-on manager to register the changes, performed FreeCAD restart
4.  activated the Arch workbench
5.  I opened the menu Arch -> Rebar tools

I did not notice any changes.

![Peek 2023-07-30 21-41](https://github.com/FreeCAD/FreeCAD-translations/assets/39754351/4d337860-b99b-4d57-b1a6-31f4408b15c9)

**Comment by luzpaz** [comment]
> Will you be doing a string import into Crowdin?

@amrit3701 this is a key step, was this done for the Reinforcement WB ?

**Comment by kaktusus** [comment]
fixed
pre-translation :wink:

![obraz](https://github.com/FreeCAD/FreeCAD-translations/assets/39754351/d8da13ed-7277-4b44-8c75-0b8a63b0c708)

**Comment by kaktusus** [comment]
fixed
pre-release translation :wink:

![obraz](https://github.com/FreeCAD/FreeCAD-translations/assets/39754351/d8da13ed-7277-4b44-8c75-0b8a63b0c708)
